src/main.c

```diff
@@ -1,25 +1,53 @@
 #include <stdlib.h>
 #include <time.h>
+#include <stdio.h>
 
 #include "../include/raylib.h"
 #include "../include/raymath.h"
 
+#include "tmp_structs.h"
 
 int screenWidth = 800;
-int screenHeight = 480;
-char title[] = "Raylib Template";
+int screenHeight = 800;
+char *title = "Minesweeper-RTS";
 
 int main() {
+    StateManager *states = (StateManager*) malloc(sizeof(StateManager));
+
     srand(time(0));
     InitWindow(screenWidth, screenHeight, title);
 
+    states->state=0;//0->start 1->start game settings 2->game 3->settings 4-> statistics 5->quit
+
+
+    SetTargetFPS(60);
+
+    
     while (!WindowShouldClose()) {
+
         BeginDrawing();
         ClearBackground(RAYWHITE);
         DrawText("Example Window!", screenWidth/2 - MeasureText("Example Window!", 32) / 2, screenHeight / 2 - 32 / 2, 32, DARKGRAY);
+        if(states->state==0){
```

Comment:
Let's not add integer literals like `0`; let's declare some constants in a constants file instead. Also, since there is only one state right now, let's only account for this state. The rest of the state with the comments can be added as necessary. Let's also change this to a `switch` statement if possible.

Note: The default of the `switch` statement can just be an error thrown but for now, let's just add a `// TODO` over there and handle errors later.

Reply:
I have add the switch statement and used a header file to specify constant variables for the state